sakiparens.py

```diff
@@ -0,0 +1,17 @@
+from __future__ import unicode_literals
+
+def parens(string):
+    stack = []
+    for char in string:
+        if char == '(':
+            stack.append(char)
+        elif char == ')':
+            try:
+                stack.pop()
```

Comment:
`.pop()` on a list is an expensive operation (O(n)).  You can do much better by using an actual stack where both insert and pop are O(1)